### DISCUSSION TITLE
Can I download a PDF file with Flaresolverr

### DISCUSSION THREAD

**Discussion by paulvav** [FlareSolverr/FlareSolverr#865] [Q&A]
Was wondering if its possible to download a full pdf file via flaresolverr. Im currently scraping a site and need to get past their cloudflare protection to get and download pdf files. I have no problem scraping the rest of their site with flaresolverr - but unsure how to go about grabbing the contents of a file.

This is how I normally download and store a PDF file in python without flaresolverr:

```
#...
  req = requests.get(url, verify=False)

  with open(uid, "wb") as outfile:
      outfile.write(req.content)

```

At the moment the response from flaresolverr from the protected url seems to contain the HTML of the page - but has no "req.content" equivalent with the actual file contents.

Really appreciate any pointers anyone could give me on this.

**Comment by Charlotte-br560** [comment]
Certainly! To download a PDF file while utilizing Flaresolverr to bypass Cloudflare protection, consider integrating Flaresolverr with the requests library. Explore how similar tasks are managed in other projects for potential insights and recommendations. This can aid in efficiently retrieving the actual file contents and adapting your current implementation for PDF file extraction. Best of luck with your scraping efforts!
